packages/headless/src/controllers/commerce/product-listing/headless-product-listing.test.ts

```diff
@@ -57,8 +52,8 @@ describe('headless product-listing', () => {
 
     expect(buildProductListingSubControllers).toHaveBeenCalledWith(engine, {
       responseIdSelector,
-      fetchProductsActionCreator: ProductListingActions.fetchProductListing,
-      fetchMoreProductsActionCreator: ProductListingActions.fetchMoreProducts,
```

Comment:
Why? It seems we are weakening the test here?

Reply:
This checked that the exact same function reference was passed. But now we're passing closures that are created fresh each time, so reference equality fails.

However, I could extend the tests further to use mocks to verify that the closures are capturing the correct `enableResults` value. I'll update